wire-library/wire-runtime/src/commonMain/kotlin/com/squareup/wire/Duration.kt

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire
+
+expect class Duration {
+  fun getSeconds(): Long
+  fun getNano(): Int
+}
+
+expect fun durationOfSeconds(seconds: Long, nanoAdjustment: Long): Duration
```

Comment:
nanos should be `Int` not `Long`.

Also, this seems like it wants to be
```kotlin
fun Duration(seconds: Long, nanoAdjustment: Int)
```
so we get constructor-like behavior. Because it's not really "of seconds" since we can specify nanos as well. Plus maybe we specify nanos to be 0 by default?

Reply:
Ah crap the Java method is named "of seconds"...

Reply:
and `ofSeconds` takes both as longs :O

Reply:
Weird. I suspect that's because they want to be friendly to people using `nanoTime()` and doing math? What happens if you exceed a second's worth of nanos, I wonder.